### PR TITLE
Nn 2038 change column names on activity list

### PR DESCRIPTION
**Pull request by JamieDraperUK**
**Integration tests are coming in the next commit as this one is getting fairly big and wanted to commit early to avoid conflicts**

- Change column names on the activity lists
- Removed the stars from various events for example *** Transfer (cancelled) ***
- Rework absent reason data structure so that we can change the presentation of the reasons but not change the id
- Move the absent reason mapping logic into a central location

### CODE REVIEW

File: src/OtherActivityListView/index.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ const renderOtherEvent = (event, index) => {
...
 const renderEvent = (event, type, index) => (
   <li className="transfer" key={`${type}_${index}`}>
-    <strong className="other-activity">** {event.eventDescription} ** </strong>
-    {event.expired && <span className="cancelled">(expired)</span>}
-    {event.complete && <span className="complete">(complete)</span>}
-    {event.cancelled && <span className="cancelled">(cancelled)</span>}
+    <strong className="other-activity">{event.eventDescription}</strong>
+    {event.expired && <span className="cancelled">{`${' (expired)'}`}</span>}
+    {event.complete && <span className="complete">{`${' (complete)'}`}</span>}
+    {event.cancelled && <span className="cancelled">{`${' (cancelled)'}`}</span>}
```
Comment (jsrobertson) [inline src/OtherActivityListView/index.js]: Do these need to be template literals?

File: src/ResultsActivity/ResultsActivity.js [hunk 2]
```diff
@@ -165,8 +165,8 @@ class ResultsActivity extends Component {
           name={['updateAttendanceEnabled']}
           render={() => (
             <React.Fragment>
-              <th className="straight width5 no-print">Pay</th>
-              <th className="straight width5 no-print">Other</th>
+              <th className="straight width4 no-print">Attended</th>
+              <th className="straight width6 no-print">Not attended</th>
```
Comment (jsrobertson) [inline src/ResultsActivity/ResultsActivity.js]: 2058 mentions Non-attendance. Which ones right? :D

File: src/ResultsActivity/elements/PayOptions/PayOptions.js [hunk 3]
```diff
@@ -79,17 +78,18 @@ function PayOptions({ offenderDetails, updateOffenderAttendance, date, openModal
...
       )}
-
       <Option data-qa="other-option" className="row-gutters">
-        {showRadioButton && (
-          <Radio name={offenderNo} onChange={renderForm} value="other" checked={selectedOption === 'other'}>
-            <VisuallyHidden>Other</VisuallyHidden>
-          </Radio>
-        )}
+        {showRadioButton &&
+          !absentReason && (
```
Comment (jsrobertson) [inline src/ResultsActivity/elements/PayOptions/PayOptions.js]: Should `!absentReason` be covered by `showRadioButton`?

File: src/ResultsActivity/elements/PayOptions/PayOptions.js [hunk 4]
```diff
@@ -79,17 +78,18 @@ function PayOptions({ offenderDetails, updateOffenderAttendance, date, openModal
...
+            <Radio name={offenderNo} onChange={renderForm} value="other" checked={selectedOption === 'other'}>
+              <VisuallyHidden>Other</VisuallyHidden>
+            </Radio>
+          )}
         {allowUpdate &&
-          other && (
-            <UpdateLink onClick={renderForm}>
-              <OtherMessage data-qa="other-message">View/ Update</OtherMessage>
+          other &&
+          absentReason && (
```
Comment (jsrobertson) [inline src/ResultsActivity/elements/PayOptions/PayOptions.js]: `allowUpdate` might be better containing `other` and `absentReason` too

File: src/ResultsActivity/elements/PayOptions/PayOptions.js (hunk 3 above)
Comment (JamieDraperUK) [reply src/ResultsActivity/elements/PayOptions/PayOptions.js]: I like it!

File: src/ResultsActivity/elements/PayOptions/PayOptions.js (hunk 4 above)
Comment (JamieDraperUK) [reply src/ResultsActivity/elements/PayOptions/PayOptions.js]: I like it!

**Pull request merged**